plugins/uefi-pk/fu-uefi-pk-device.c

```diff
@@ -0,0 +1,250 @@
+		if (g_getenv("FWUPD_UEFI_PK_VERBOSE") != NULL)
+			g_debug("PK subject: %s", subject->data);
+		if (!fu_uefi_pk_device_parse_buf(self,
+						 (const gchar *)subject->data,
+						 subject->size,
+						 error))
+			return FALSE;
+	}
+
+	/* key ID */
+	rc = gnutls_x509_crt_get_key_id(crt, 0, key_id, &key_idsz);
+	if (rc < 0) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_INVALID_DATA,
+			    "failed to get key ID: %s [%i]",
+			    gnutls_strerror(rc),
+			    rc);
+		return FALSE;
+	}
+	key_idstr = g_compute_checksum_for_data(G_CHECKSUM_SHA1, key_id, key_idsz);
+	if (key_idstr == NULL) {
+		g_set_error(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_INVALID_DATA,
+			    "failed to calculate key ID for 0x%x bytes",
+			    (guint)key_idsz);
+		return FALSE;
+	}
+	fu_device_add_instance_strup(FU_DEVICE(self), "CRT", key_idstr);
+
+	/* success, certificate was parsed correctly */
+	return fu_device_build_instance_id(FU_DEVICE(self), error, "UEFI", "CRT", NULL);
+}
+
+static gboolean
+fu_uefi_pk_device_probe(FuDevice *device, GError **error)
+{
+	FuUefiPkDevice *self = FU_UEFI_PK_DEVICE(device);
+	g_autoptr(FuFirmware) img = NULL;
+	g_autoptr(FuFirmware) pk = fu_efi_signature_list_new();
+	g_autoptr(GBytes) pk_blob = NULL;
+	g_autoptr(GPtrArray) sigs = NULL;
+
+	pk_blob = fu_efivar_get_data_bytes(FU_EFIVAR_GUID_EFI_GLOBAL, "PK", NULL, error);
+	if (pk_blob == NULL)
+		return FALSE;
+	if (!fu_firmware_parse(pk, pk_blob, FU_FIRMWARE_FLAG_NONE, error)) {
+		g_prefix_error(error, "failed to parse PK: ");
+		return FALSE;
+	}
+
+	/* by checksum */
+	img = fu_firmware_get_image_by_checksum(pk, FU_UEFI_PK_CHECKSUM_AMI_TEST_KEY, NULL);
+	if (img != NULL)
+		self->has_pk_test_key = TRUE;
+
+	/* by text */
+	sigs = fu_firmware_get_images(pk);
+	for (guint i = 0; i < sigs->len; i++) {
+		FuEfiSignature *sig = g_ptr_array_index(sigs, i);
+		if (!fu_uefi_pk_device_parse_signature(self, sig, error))
+			return FALSE;
+	}
+
+	/* success */
+	return TRUE;
+}
+
+static void
+fu_uefi_pk_device_add_security_attrs(FuDevice *device, FuSecurityAttrs *attrs)
+{
+	FuUefiPkDevice *self = FU_UEFI_PK_DEVICE(device);
+	g_autoptr(FwupdSecurityAttr) attr = NULL;
+
+	/* create attr */
+	attr = fu_device_security_attr_new(device, FWUPD_SECURITY_ATTR_ID_UEFI_PK);
+	fu_security_attrs_append(attrs, attr);
+
+	/* test key is not secure */
+	if (self->has_pk_test_key) {
+		fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_NOT_VALID);
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_ACTION_CONFIG_FW);
+		fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_ACTION_CONTACT_OEM);
+		return;
+	}
+
+	/* success */
+	fwupd_security_attr_add_flag(attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+	fwupd_security_attr_set_result(attr, FWUPD_SECURITY_ATTR_RESULT_VALID);
+}
+
+static void
+fu_uefi_pk_device_init(FuUefiPkDevice *self)
+{
+	fu_device_set_physical_id(FU_DEVICE(self), "pk");
+	fu_device_set_name(FU_DEVICE(self), "UEFI pk");
```

Comment:
Although correct this is going to mean squat to most people.  How about `UEFI Platform Key`?  And then describe in the description what it is